eslint.config.mjs

```diff
@@ -0,0 +1,467 @@
+import globals from "globals";
+
+import import_ from "eslint-plugin-import";
+import jasmine from "eslint-plugin-jasmine";
+import json from "eslint-plugin-json";
+import noUnsanitized from "eslint-plugin-no-unsanitized";
+import perfectionist from "eslint-plugin-perfectionist";
+import prettierRecommended from "eslint-plugin-prettier/recommended";
+import unicorn from "eslint-plugin-unicorn";
+
+const jsFiles = folder => {
+  const prefix = folder === "." ? "" : folder + "/";
+  return [prefix + "**/*.js", prefix + "**/*.jsm", prefix + "**/*.mjs"];
+};
+
+// Include all files referenced in extensions/chromium/background.js
+const chromiumExtensionServiceWorkerFiles = [
+  "extensions/chromium/options/migration.js",
+  "extensions/chromium/preserve-referer.js",
+  "extensions/chromium/pdfHandler.js",
+  "extensions/chromium/extension-router.js",
+  "extensions/chromium/suppress-update.js",
+  "extensions/chromium/telemetry.js",
+];
+
+export default [
+  {
+    ignores: [
+      "**/build/",
+      "**/l10n/",
+      "**/docs/",
+      "**/node_modules/",
+      "external/bcmaps/",
+      "external/builder/fixtures/",
+      "external/builder/fixtures_babel/",
+      "external/quickjs/",
+      "external/openjpeg/",
+      "test/tmp/",
+      "test/pdfs/",
+      "web/locale/",
+      "**/*~/",
+    ],
+  },
+
+  /* ======================================================================== *\
+                             Base configuration
+  \* ======================================================================== */
+
+  prettierRecommended,
+  {
+    files: ["**/*.json"],
+    ...json.configs.recommended,
+  },
+  {
+    files: jsFiles("."),
+    ignores: chromiumExtensionServiceWorkerFiles,
+    languageOptions: {
+      globals: globals.browser,
+    },
+  },
+  {
+    files: jsFiles("."),
+
+    plugins: {
+      import: import_.flatConfigs.recommended.plugins.import,
+      json,
+      "no-unsanitized": noUnsanitized,
+      perfectionist,
+      unicorn,
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.worker,
+        PDFJSDev: "readonly",
+        __non_webpack_import__: "readonly",
+      },
+
+      ecmaVersion: 2022,
+      sourceType: "module",
+    },
+
+    rules: {
+      "import/export": "error",
+      "import/exports-last": "error",
+
+      "import/extensions": ["error", "always", { ignorePackages: true }],
+
+      "import/first": "error",
```

Comment:
Here, and in lots of other spots below, why are additional blank lines now inserted between *some* of the rules?

Reply:
Oh I used a tools to convert the original top-level config (`npx @eslint/migrate-config .eslintrc`) and I didn't notice it generated these lines. Removed.